msg/RadarDetection.msg

```diff
@@ -0,0 +1,10 @@
+# All variables below are relative to the radar's frame of reference.
+# This message is not meant to be used alone but as part of a stamped or array message.
+
+# Radar detections are the discrete clusterings of points
+# Detections which are filtered into moving objects are referred to as Tracks
+
+                                            # Note: The z component of these fields is ignored for 2D tracking.
+geometry_msgs/Point position                # x, y, z coordinates of the centroid of the object being tracked.
+geometry_msgs/Vector3 velocity              # The velocity of the object in each spatial dimension.
+geometry_msgs/Vector3 size                  # The object size (m) in the x, y, z, dimensions
```

Comment:
Is it even possible to actually extract 3D dimension information from a radar detection? Isn't it more like 2D dimension what you see in the radar? Because of occlusions, you can probably never correctly estimate "depth", you only have its lower bound.

Reply:
You can get 3D positional data from Radar (at least with FMCW Radar). It really comes down to the configuration of the antennas. Radar can estimate the reflection in the Z plane in a similar way to the X plane. 

You are right in that radar works on reflections so can't really access the 3Dness of solid objects, however depending on the object you can get some data out. Each type of object as a kind of "depth signature" based on way it interacts with the reflections, and thin objects (like glass) cause a partial reflection while still allowing other objects behind to be detected.